lib/DBIx/Connector/Pg.pm

```diff
@@ -0,0 +1,60 @@
+package DBIx::Connector::Pg;
+# ABSTRACT: ...
+
+use strict;
+use warnings;
+use parent 'DBIx::Connector';
+
+our $VERSION = '0.001';
+
+sub connected {
+    my $self = shift;
+    return unless $self->_seems_connected;
+    my $dbh = $self->{_dbh} or return;
+    $dbh->state and $dbh->state =~ /^08|^57/ and return;
+    return 1;
+    #dbh->state and $dbh->state =~ /^BI/ and return 1;
+    #return $self->driver->ping($dbh);
+}
```

Comment:
Here is the 3rd thought. You are right to include the error class `57`. Well, except `57014`.

```perl
$dbh->state and $dbh->state =~ /^08|^57(?!014)/ and return;
```